packages/@aws-cdk/aws-sagemaker/lib/notebook-instance.ts

```diff
@@ -0,0 +1,237 @@
+import ec2 = require('@aws-cdk/aws-ec2');
+import iam = require('@aws-cdk/aws-iam');
+import kms = require('@aws-cdk/aws-kms');
+import { Construct, Fn, Lazy, Resource } from '@aws-cdk/core';
+import { CfnNotebookInstance, CfnNotebookInstanceLifecycleConfig } from './sagemaker.generated';
+
+/**
+ * Construction properties for a SageMaker Notebook instance
+ *
+ * @experimental
+ */
+export interface NotebookInstanceProps {
+
+    /**
+     * Enable the notebook to have direct internet access.
+     *
+     * @default true
+     */
+    readonly enableDirectInternetAccess?: boolean;
+
+    /**
+     * Instance type of the notebook.
+     *
+     * @default m1.t2.medium
+     */
+    readonly instanceType?: ec2.InstanceType;
+
+    /**
+     * Encryption key for the EBS volume attached to the notebook instance.
+     *
+     * @default none
+     */
+    readonly kmsKeyId?: kms.IKey;
+
+    /**
+     * Name of the notebook instance.
+     *
+     * @default none
+     */
+    readonly notebookInstanceName?: string;
+
+    /**
+     * Role to provide to the Sagemaker service to access other AWS services.
+     *
+     * @default a new role for Amazon SageMaker.
+     */
+    readonly role?: iam.IRole;
+
+    /**
+     * Enable the root access to the notebook instance.
+     *
+     * @default true
+     */
+    readonly enableRootAccess?: boolean;
+
+    /**
+     * VPC to deploy the notebook instance.
+     *
+     * @default none
+     */
+    readonly vpc?: ec2.IVpc;
+
+    /**
+     * Subnet where the notebook instance is deployed to in the VPC.
+     *
+     * @default none
+     */
+    readonly subnet?: ec2.ISubnet;
+
+    /**
+     * Size of the notebook volume in GB.
+     *
+     * @default 5 GB
+     */
+    readonly volumeSizeInGB?: number;
+}
+
+/**
+ * Defines a SageMaker Notebook instance.
+ *
+ * @experimental
+ */
+export class NotebookInstance extends Resource implements ec2.IConnectable {
+
+    /**
+     * Allows specify security group connections for instances of this fleet.
+     */
+    public readonly connections: ec2.Connections;
+
+    /**
+     * Notebook instance name.
+     *
+     * @attribute
+     */
+    public readonly notebookInstanceName: string;
+
+    private readonly vpc: ec2.IVpc;
+    private readonly securityGroup: ec2.ISecurityGroup;
+    private readonly securityGroups: ec2.ISecurityGroup[] = [];
+    private readonly subnet: ec2.ISubnet;
+
+    private readonly role: iam.IRole;
+    private readonly instanceType: string;
+    private readonly onCreateLines = new Array<string>();
+    private readonly onStartLines = new Array<string>();
+    private readonly lifecycleConfigName: string;
+
+    constructor(scope: Construct, id: string, props: NotebookInstanceProps = {}) {
+        super(scope, id);
```

Comment:
You need to pass the `props.notebookInstanceName` here as the value of the `physicalName` property of the third argument to the `super` call.